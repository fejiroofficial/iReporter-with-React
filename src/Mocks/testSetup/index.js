/* eslint-disable import/no-extraneous-dependencies */
import '@babel/polyfill';
import React from 'react';
import { Provider } from 'react-redux';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import configMockStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';


export const mockState = {
  user: {
    token: '',
    isLoggedIn: false,
    loginStatus: false,
    message: '',
    isLoading: false,
    profile: {
      firstname: 'John',
      username: 'johnny',
      image: 'image.jpg',
      errorData: [],
    },
    resetPassword: {
      loading: false,
      message: '',
    }
  },
  bookmark: {
    loading: false,
    articles: [],
    message: '',
  },
  article: {
    newComments: {
      body: 'Hello world',
      id: 1,
      authorName: 'John',
      authorImage: 'image.jpg',
      date: 'July 4th 2018',
    },
    postingComment: false,
    commentMessage: 'Done',
    gettingComments: false,
    oldComments: [],
    hasMoreComments: false,
    commentPage: {
      current: 1,
      currentCount: 1,
      totalCount: 1,
    },
  },
  createArticle: {
    loading: false,
    message: false,
  },
  articlesReducer: {
    articles: []
  },
  articlesCategoryReducer: []
};


export const createMockStore = () => {
  const mockStore = configMockStore([thunk]);
  configure({ adapter: new Adapter() });
  return mockStore;
};

const setup = (component, initailState = mockState) => {
  const mockStore = configMockStore([thunk]);
  configure({ adapter: new Adapter() });
  const connectedWrapper = mount(
    <Provider store={mockStore(initailState)}>
      <BrowserRouter>
        {component}
      </BrowserRouter>
    </Provider>
  );
  return connectedWrapper;
};

export default setup;
