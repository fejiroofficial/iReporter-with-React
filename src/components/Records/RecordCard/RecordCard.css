.RecordCard {
  display: block;
  height: 313px;
  width: 301px;
  margin-left: 10px;
  display: flex;
  flex-wrap: wrap;
  padding-left: 9px;
  box-shadow: 0px 0px 10px 1px lightgrey;
  padding-top: 7px;
  margin-bottom: 16px;
}

.IconHolder {
  height: fit-content;
  width: 383px;
}

.RedFlagIcon {
  height: 23px;
  width: 21px;
  background-image: url('../../../assets/img/red_flag_13.png');
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  float: right;
}

.Details {
  float: right;
}

.DetailsHolder {
  height: fit-content;
  width: 370px;
}

.NameDateDiv{
  margin-left: 10px;
}

.NameDiv {
  font-weight: bold;
}

.DateDiv {
  font-size: 12px;
  width: 138px;
}

.RecordCard .Comment {
  min-height: 178px;
  width: 301px;
  padding: 10px;
  overflow-wrap: break-word;
  margin-left: 51px;
  margin-top: 20px;
  font-size: 14px;
}

.InfoPanel {
  height: 50px;
  width: fit-content;
  display: flex;
}

.CardFooter {
  display: flex;
  margin-left: 10px;
  font-size: 12px;
  font-weight: bold;
}

.CardFooter button {
  background-color: lightseagreen;
  color: white;
  height: inherit;
  outline: none;
  border-width: inherit;
  border-radius: 3px;
  border-color: none;
  cursor: pointer;
}

.StatusHolder {
  width: 200px;
} 

@media (min-width: 360px) {
  .RecordCard {
    display: block;
    margin-left: 30px;
  }
}

@media (min-width: 375px) {
  .RecordCard {
    display: block;
    padding-left: 24px;
    width: 340px;
    margin-left: 16px;
  }
}

@media (min-width: 414px) {
  .RecordCard {
    display: block;
    padding-left: 24px;
    width: 340px;
    margin-left: 38px;
  }
}

@media (min-width: 768px) {
  .RecordCard {
    display: block;
    padding-left: 15px;
    padding-top: 19px;
    width: 615px;
    min-height: 180px;
    margin-left: 91px;
  }
}

@media (min-width: 1024px) {
  .RecordCard {
    display: block;
    padding-left: 15px;
    margin-left: 209px;
  }
}

@media (min-width: 1359px) {
  .RecordCard {
    display: block;
    margin-left: 423px;
  }
}

@media (min-width: 1583px) {
  .RecordCard {
    display: block;
    margin-left: 504px;
  }
}