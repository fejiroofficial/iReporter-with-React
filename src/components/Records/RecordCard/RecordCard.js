import React from 'react';
import Classes from './RecordCard.css';
import DisplayPicture from '../../DisplayPicture/DisplayPicture';

const recordCard = (props) => (
  <div className={Classes.RecordCard}>
    <div className={Classes.InfoPanel}>
      <DisplayPicture image={props.image}/>
      <div className={Classes.NameDateDiv}>
        <div className={Classes.NameDiv}>{`${props.firstname} ${props.lastname}`}</div>
        <div className={Classes.DateDiv}>{props.date}</div>
      </div>
      <div className={Classes.IconHolder}>
        <div className={Classes.RedFlagIcon} />
      </div>
    </div>
    <div className={Classes.Comment}>{props.comment}</div>
    <div className={Classes.CardFooter}>
      <div className={Classes.StatusHolder}>Status: {props.status}</div>
      <div className={Classes.DetailsHolder}>
        <div className={Classes.Details}>For more details click <button>here</button></div>
      </div>
    </div>
  </div>
);

export default recordCard;
