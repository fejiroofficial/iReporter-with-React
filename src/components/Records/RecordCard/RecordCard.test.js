import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { createMockStore } from '../../../Mocks/testSetup';
import RecordCard from './RecordCard';

createMockStore();

describe('<RecordCard>', () => {
  it('should render without crashing', () => {
    const container = (
      <RecordCard>
        <p>Hello</p>
      </RecordCard>
    );
    const wrapper = shallow(container);
    expect(wrapper).toMatchSnapshot();
  });
});
