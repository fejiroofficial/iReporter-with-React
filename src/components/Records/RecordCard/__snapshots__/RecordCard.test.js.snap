// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecordCard> should render without crashing 1`] = `
<Link
  replace={false}
  to="/user/ireport/undefineds/undefined"
>
  <div>
    <div
      style={
        Object {
          "backgroundImage": "url(undefined)",
          "backgroundPosition": "50%",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
          "height": "150px",
          "width": "inherit",
        }
      }
    />
    comment:
    <div />
    <div>
      <displayPicture />
      <div>
        <div>
          undefined undefined
        </div>
        <div />
      </div>
      <div>
        <div />
      </div>
    </div>
    <div>
      <div>
        Status: 
      </div>
    </div>
  </div>
</Link>
`;
